docs/proposals/console/devfile-import.md

```diff
@@ -0,0 +1,77 @@
+# Console Import from Devfile
+
+This document outlines how the console is going to use a devfile 2.0.0 spec for it's import feature targeted for the upcoming release.
+
+## As-Is Today
+
+Currently the Devfile import feature is mocked with a POC [openshift/console PR](https://github.com/openshift/console/pull/6321). The POC PR, requires the build guidance devfile spec to be implemented. However, the build guidance spec is still an open discussion in the [devfile/api PR](https://github.com/devfile/api/pull/127). 
+
+With an initial target date for the Dec 04, 2020; the devfile import developer preview should look similar to this [demo video](https://drive.google.com/file/d/1uLzDibVZlkMqbjtKkho04e8k2-Ns5A2W/view).
+
+The information required to build a component from build guidance are:
+- Git Repo Url
+- Git Repo Ref
+- Build Context Path
+- Container Port
+- Image Name
+
+The console devfile import page has the Git Repo Url, Git Repo Ref & the Build Context Path
+
+<img src="https://user-images.githubusercontent.com/31771087/99319303-4ae89180-2837-11eb-8933-eaaf41160bcd.png">
+
+The Dockerfile is assumed to be present in the context directory. The Container Port is assumed to be 8080 and the Image Name is assumed to be the same as the application name.
+
+### Note:
+The context path is present in both the Console import form and the open devfile spec(Refer to the OpenShift Console Devfile Import screenshot above and the open Build Guidance PR screenshot below). Do we require two context dir information? Is the context dir in the Console form for devfile.yaml and the context dir in the build guidance spec for Dockerfile relative to devfile.yaml? Or do we always assume both the devfile.yaml and Dockerfile are always at the same dir level.
+
+ For phase 1, only the Dockerfile Build Guidance is to be implemented. SourceToImage(S2I) will be implemented in the future sprints.
+
+<img src="https://user-images.githubusercontent.com/31771087/99319306-4c19be80-2837-11eb-9639-a5c130deb4ba.png">
+
+## Proposed Changes
+
+With the target deadline approaching soon, we may need to provide an alternative path for the OpenShift Console to consume the devfile. With the Build Guidance spec [devfile/api PR](https://github.com/devfile/api/pull/127) still an open discussion, this document proposes a solution within the boundaries of the devfile 2.0.0 spec to achieve this.
+
+Here is a proposed 2.0.0 spec devfile that can support build guidance for OpenShift Console:
```

Comment:
Can you make it clear what exactly your proposal will add/change? It's a little hard to tell what's changing, apart from the `alpha.build-dockerfile` attribute.

Reply:
That is perhaps the only change without the build guidance spec in the current devfile spec. I'll be more clear!

Reply:
Okay, that makes sense.

Reply:
Updated